libs/shared/shared/django_apps/upload_breadcrumbs/admin.py

```diff
@@ -519,3 +545,275 @@ def has_add_permission(self, request: HttpRequest, obj: Any = None) -> bool:
 
     def has_change_permission(self, request: HttpRequest, obj: Any = None) -> bool:
         return False
+
+    def get_urls(self):
+        urls = super().get_urls()
+        custom_urls = [
+            path(
+                "<path:object_id>/resend-upload/",
+                self.admin_site.admin_view(self.resend_upload_view),
+                name="upload_breadcrumbs_uploadbreadcrumb_resend_upload",
+            ),
+        ]
+        return custom_urls + urls
+
+    @admin.display(description="Actions", ordering=None)
+    def resend_upload_button(self, obj: UploadBreadcrumb) -> str:
+        """Display resend button in the list view for failed uploads."""
+        if not self._is_failed_upload(obj):
+            return "-"
+
+        resend_url = reverse(
+            "admin:upload_breadcrumbs_uploadbreadcrumb_resend_upload", args=[obj.id]
+        )
+        return format_html(
+            '<a class="button" href="{}" onclick="return confirm(\'Are you sure you want to resend this upload?\')">🔄 Resend</a>',
+            resend_url,
+        )
+
+    @admin.display(description="Resend Upload")
+    def resend_upload_action(self, obj: UploadBreadcrumb) -> str:
+        """Display resend actions in the detail view."""
+        if not obj.pk:  # New object
+            return "-"
+
+        html_parts = []
+
+        if self._is_failed_upload(obj):
+            resend_url = reverse(
+                "admin:upload_breadcrumbs_uploadbreadcrumb_resend_upload", args=[obj.id]
+            )
+            html_parts.append(
```

Comment:
Is there a way to only append 1 entry here? Or what's the intent w/ having multiple `html_parts.append`?